commercetools-convenience/src/test/java/io/sphere/sdk/meta/CategoryDocumentationIntegrationTest.java

```diff
@@ -73,29 +74,110 @@ public static void beforeClass() throws IOException {
+            .queryAll(client(), CategoryQuery.of(), externalIdCallBack, 4);
+        final List<List<String>> categoryPages = SphereClientUtils
+            .blockingWait(categoryPagesStage, Duration.ofMinutes(5));
+
+        assertThat(categoryPages).hasSize(4);
+
+        final List<String> externalIds = categoryPages.stream()
+                                                      .flatMap(List::stream)
+                                                      .collect(toList());
+
+        assertThat(externalIds).hasSize(15);
+        IntStream.range(0, externalIds.size()).forEach(index -> assertThat(externalIds).contains(index + ""));
+    }
+
+    @Test
+    public void collectAllExternalIds() throws Exception {
+        final List<String> externalIds = new ArrayList<>();
+
+        final Consumer<List<Category>> categoryPageConsumer = categoriesPage ->
+            categories.forEach(category -> externalIds.add(category.getExternalId()));
+
+        final CompletionStage<Void> categoryPagesStage = QueryExecutionUtils
+            .queryAll(client(), CategoryQuery.of(), categoryPageConsumer, 500);
+        SphereClientUtils.blockingWait(categoryPagesStage, Duration.ofMinutes(5));
+
+        assertThat(externalIds).hasSize(15);
+        IntStream.range(0, externalIds.size()).forEach(index -> assertThat(externalIds).contains(index + ""));
+    }
+
     @Test
     public void fetchAllAsJson() throws Exception {
-        final CompletionStage<List<JsonNode>> categoriesStage = QueryExecutionUtils.queryAll(client(), JsonNodeQuery.of("/categories"), 500);;
-        final List<JsonNode> categories = SphereClientUtils.blockingWait(categoriesStage, Duration.ofMinutes(5));
+        final CompletionStage<List<List<JsonNode>>> categoryPagesStage = QueryExecutionUtils
+            .queryAll(client(), JsonNodeQuery.of("/categories"), (categories -> categories), 500);
+        final List<List<JsonNode>> categoryPages = SphereClientUtils
+            .blockingWait(categoryPagesStage, Duration.ofMinutes(5));
+        final List<JsonNode> categories = categoryPages.stream()
+                                                       .flatMap(List::stream)
+                                                       .collect(toList());
+
         assertThat(categories)
-                .hasSize(15)
-                .matches(cats -> cats.parallelStream().anyMatch(cat -> cat.get("slug").get("en").asText().equals("boots-women")));
+            .hasSize(15)
+            .matches(cats -> cats.parallelStream()
+                                 .anyMatch(cat -> cat.get("slug").get("en").asText().equals("boots-women")));
     }
 
     @Test
     public void fetchRoots() throws Exception {
         final CategoryQuery seedQuery = CategoryQuery.of().withPredicates(m -> m.parent().isNotPresent());
-        final CompletionStage<List<Category>> categoriesStage = QueryExecutionUtils.queryAll(client(), seedQuery);
```

Comment:
When you changed the code to also provide the old API, you should be able to revert these changes.

Reply:
reverted legacy API: 16d5ea6